qutebrowser/keyinput/basekeyparser.py

```diff
@@ -366,6 +365,7 @@ def on_keyconfig_changed(self, mode):
 
     def clear_keystring(self):
         """Clear the currently entered key sequence."""
-        self._debug_log("discarding keystring '{}'.".format(self._keystring))
-        self._keystring = ''
-        self.keystring_updated.emit(self._keystring)
+        if self._keystring:
+            self._debug_log("discarding keystring '{}'.".format(self._keystring))
+            self._keystring = ''
+            self.keystring_updated.emit(self._keystring)
```

Comment:
I think it should work with the strict checks, but doing that right would mean careful thinking in the subclasses. So if you don't mind, I'd leave it this way for now and return to it with next PR (I have some minor refactoring in mind for `HintKeyParser` anyway).